homeassistant/components/litterrobot/vacuum.py

```diff
@@ -0,0 +1,126 @@
+"""Support for Litter-Robot "Vacuum"."""
+from pylitterbot import Robot
+
+from homeassistant.components.vacuum import (
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_ERROR,
+    SUPPORT_SEND_COMMAND,
+    SUPPORT_START,
+    SUPPORT_STATE,
+    SUPPORT_STATUS,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    VacuumEntity,
+)
+from homeassistant.const import STATE_OFF
+
+from .const import DOMAIN
+from .hub import LitterRobotEntity
+
+SUPPORT_LITTERROBOT = (
+    SUPPORT_SEND_COMMAND
+    | SUPPORT_START
+    | SUPPORT_STATE
+    | SUPPORT_STATUS
+    | SUPPORT_TURN_OFF
+    | SUPPORT_TURN_ON
+)
+TYPE_LITTER_BOX = "Litter Box"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Litter-Robot cleaner using config entry."""
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for robot in hub.account.robots:
+        entities.append(LitterRobotCleaner(robot, TYPE_LITTER_BOX, hub))
+
+    if entities:
+        async_add_entities(entities, True)
+
+
+class LitterRobotCleaner(LitterRobotEntity, VacuumEntity):
```

Comment:
This is really pushing the definition of a vacuum, but I don't have a better suggestion.

Reply:
@bdraco, I agree it stretches the concept of "vacuum". It's more of a "cleaner" (dishwasher, washing machine, etc) or more generically an "appliance" (fridge, dryer, etc). Maybe even just "robot"? But that would be an entirely new platform and wasn't sure that I'm capable of thinking through all the possibilities at the moment, haha.